nservicebus/messaging/message-headers-for-third-party-integrations.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Required Headers For Third Party Integration
+summary: Lists the necessary headers when receiving messages from non-NSB endpoints for better monitoring.
+tags:
+- Monitoring
+- Third Party Integration
+- ServiceInsight
+- ServicePulse
+related:
+- nservicebus/messaging/message-headers
+---
+
+# How to integrate external systems with NServiceBus?
+
+When an NServiceBus endpoint receives messages from external systems such as BizTalk, TIBCO, etc directly, the message itself will not contain regular NServiceBus headers. Some headers are required by NServiceBus to handle those messages (i.e. enable deserialization of the message). Others are essential to provide effective monitoring and troubleshooting experience. 
+
+When NServiceBus endpoints send messages, they include a certain set of headers such as the type of message, the name of the sending endpoint, the machine name etc. To get the best use of your monitoring and debugging tools, please ensure that the following headers are being added to the actual message by the third party endpoints when sending a message. 
+
+### Mandatory Headers
+
+Required for deserializing the message:
+
+Key  | Value
+------------- | -------------
+NServiceBus.EnclosedMessageTypes  | Fully Qualified Message type including the assembly name, for e.g. IntegrationSample.Messages.Commands.ProcessOrder, IntegrationSample.Messages
```

Comment:
Is the sample you mention online somewhere?

How do you serialize the data? xml? json?

On Mon, Aug 10, 2015 at 3:19 PM, Weronika Łabaj <notifications@github.com>
wrote:

> In nservicebus/messaging/message-headers-for-third-party-integrations.md
> <https://github.com/Particular/docs.particular.net/pull/729#discussion_r36629156>
> :
>
> > +- nservicebus/messaging/message-headers
> > +---
> > +
> > +# How to integrate external systems with NServiceBus?
> > +
> > +When an NServiceBus endpoint receives messages from external systems such as BizTalk, TIBCO, etc directly, the message itself will not contain regular NServiceBus headers. Some headers are required by NServiceBus to handle those messages (i.e. enable deserialization of the message). Others are essential to provide effective monitoring and troubleshooting experience.
> > +
> > +When NServiceBus endpoints send messages, they include a certain set of headers such as the type of message, the name of the sending endpoint, the machine name etc. To get the best use of your monitoring and debugging tools, please ensure that the following headers are being added to the actual message by the third party endpoints when sending a message.
> > +
> > +### Mandatory Headers
> > +
> > +Required for deserializing the message:
> > +
> > +Key  | Value
> > +------------- | -------------
> > +NServiceBus.EnclosedMessageTypes  | Fully Qualified Message type including the assembly name, for e.g. IntegrationSample.Messages.Commands.ProcessOrder, IntegrationSample.Messages
>
> @andreasohlund <https://github.com/andreasohlund> @danielmarbach
> <https://github.com/danielmarbach> Maybe what I did in my MSMQ sample was
> not correct, but if I just send the message like this:
> var msg = new ProcessOrder { OrderId = Guid.NewGuid(), ProductId = 123456
> };
> var m = new Message(msg);
> messageQueue.Send(m, myTransaction);
> It passes type name, but excluding namespace. In such case I get
> serialization exception, because of missing namespace (NSB xml serializer
> can't find the type then in mapper).
> In RabbitMQ sample the type node contains namespace name, but can we
> assume it's always the case after serialization or should we document that
> it's required? The same goes for SQL sample.
>
> Btw if we need to check that across versions, up to which version should I
> check?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/docs.particular.net/pull/729/files#r36629156>
> .
>